Add TiDB limitations (#2899)
pingcap/docs: tidb-limitations.md

---
title: TiDB Limitations

> summary: Learn the limitations of TiDB.

Reviewer: ```suggestion
summary: Learn the usage limitations of TiDB.
```

category: introduction
---

# TiDB Limitations

> This document describes the common limitations of TiDB, including the maximum identifier length and the maximum count of supported databases, tables, indexes, partitioned tables, and sequences.

Reviewer: ```suggestion
This document describes the common usage limitations of TiDB, including the maximum identifier length and the maximum number of supported databases, tables, indexes, partitioned tables, and sequences.
```

> ## Limitation on identifier length

Reviewer: ```suggestion
## Limitations on identifier length
```

> | Identifier type | Maximum length (character) |

Reviewer: ```suggestion
| Identifier type | Maximum length (number of characters allowed) |
```

|:---------|:--------------|
| Database | 64 |
| Table    | 64 |
| Column   | 64 |
| Index    | 64 |
| View     | 64 |
| Sequence | 64 |

> ## Limitation on the total count of databases, tables, views, and connections

Reviewer: ```suggestion
## Limitations on the total number of databases, tables, views, and connections
```

> | Identifier type  | Maximum count   |

Reviewer: ```suggestion
| Identifier type  | Maximum number  |
```

|:----------|:----------|
| Databases | unlimited |
| Tables    | unlimited |
| Views     | unlimited |
| Connections| unlimited|

> ## Limitation on a single database

Reviewer: ```suggestion
## Limitations on a single database
```

| Type       | Upper limit   |
|:----------|:----------|
| Tables    | unlimited  |

> ## Limitation on a single table

Reviewer: ```suggestion
## Limitations on a single table
```

| Type       | Upper limit   |
|:----------|:----------|
| Columns   | 512       |
| Indexes    | 64        |
| Rows      | unlimited |
| Size      | unlimited |
| Partitions | 1024      |

## Limitation on a single row

| Type       | Upper limit   |
|:----------|:----------|
| Size       | 6 MB       |

## Limitation on a single column

| Type       | Upper limit   |
|:----------|:----------|
| Size       | 6 MB       |

> ## Limitation on string types

Reviewer: ditto

Reviewer: In this context, the plural form of `limitation` is more commonly used. Please pay attention to this word.

| Type       | Upper limit   |
|:----------|:----------|
| CHAR       | 256 characters      |
| BINARY     | 256 characters      |
| VARBINARY  | 65535 characters    |
| VARCHAR    | 16383 characters    |

> | TEXT       | 6MB bytes      |

Reviewer: ```suggestion
| TEXT       | 6 MB bytes      |
```

> | BLOB       | 6MB bytes      |

Reviewer: ditto

> ## Limitation on SQL statements

Reviewer: ditto

> ## Limitation on SQL statements

Reviewer: ```suggestion
## Limitations on SQL statements
```

| Type       | Upper limit   |
|:----------|:----------|

> | The number of SQL statements in a single transaction |  When the optimistic transaction is used and the transaction retry is enabled, the default upper limit is 5000, which can be changed through [`stmt-count-limit`](/tidb-configuration-file.md#stmt-count-limit). |

Reviewer: Some meaning is missed. Try to be clearer in each column.
```suggestion
| The maximum number of SQL statements in a single transaction |  When the optimistic transaction is used and the transaction retry is enabled, the default upper limit is 5000, which can be modified using [`stmt-count-limit`](/tidb-configuration-file.md#stmt-count-limit). |
```
